src/main/resources/com/atlassian/bitbucket/jenkins/internal/jenkins/oauth/servlet/AuthorizeConfirmationConfig/index.jelly

```diff
@@ -3,30 +3,42 @@
     <l:layout permission="${app.READ}">
         <l:main-panel>
-            <h2>Welcome to ${it.instanceName}</h2>
-            <p>
-                The application <strong>${it.consumerName}</strong> would like to have <strong>${it.accessRequest}</strong>
-                access to your data on <strong>${it.urlName}</strong>. The application will be allowed to use your
-                credentials and authenticate as you (<strong>${it.authenticatedUsername}</strong>) in the future.
-            </p>
-
-            <f:form action="performSubmit" method="POST" name="performSubmit">
-                <f:invisibleEntry field="oauth_token_field">
-                    <f:readOnlyTextbox type="hidden" name="${it.OAUTH_TOKEN_PARAM}" value="${it.token}"/>
-                </f:invisibleEntry>
-                <f:invisibleEntry field="oauth_callback_field">
-                      <f:readOnlyTextbox type="hidden" name="oauth_callback" value="${it.callback}"/>
-                </f:invisibleEntry>
-                <f:bottomButtonBar>
-                    <f:submit name="${it.AUTHORIZE_KEY}" value="Authorize"/>
-                    <f:submit name="${it.CANCEL_KEY}" value="Cancel"/>
-                </f:bottomButtonBar>
-            </f:form>
-
-            <p>
-                By allowing this, you can use <strong>${it.instanceName}</strong> without storing your ${it.instanceName}
-                password in <strong>${it.consumerName}</strong>.
-            </p>
+            <div align="center" style="max-width:678px; ">
+                <div style="padding-top:75px; padding-bottom:35px;">
+                    <img src="${it.iconUrl}"/>
+                </div>
+                <div>
+                    <h2>
+                        <j:choose>
+                            <j:when test="${it.getDescriptor().isAuthenticated()}">
+                                ${it.consumerName} wants to access Jenkins as ${it.authenticatedUsername}
+                            </j:when>
+                            <j:otherwise>
+                                ${it.consumerName} is requesting access to Jenkins
+                            </j:otherwise>
+                        </j:choose>
+                    </h2>
+                    <p>
+                        Clicking allow will give this Bitbucket Server instance read and write access to your data on
+                        Jenkins. This access will persist until it’s revoked.
+                    </p>
+                    <f:form action="performSubmit" method="POST" name="performSubmit">
+                        <f:invisibleEntry field="oauth_token_field">
+                            <f:readOnlyTextbox type="hidden" name="${it.OAUTH_TOKEN_PARAM}" value="${it.token}"/>
+                        </f:invisibleEntry>
+                        <f:invisibleEntry field="oauth_callback_field">
+                            <f:readOnlyTextbox type="hidden" name="oauth_callback" value="${it.callback}"/>
```

Comment:
Not your change but having ${it.CALLBACK_PARAM} instead of hardcoded value will be consistent.

Reply:
Callback param isn't a final static though, so I think it's okay to leave as is?

Reply:
don't worry about it, I have handled it in my PR.